### NO 4

Melakukan test hipotesis 1 sampel

In [30]:
import scipy.stats as st
import pandas as pd
import matplotlib.pyplot as plt
import math
from statsmodels.stats.weightstats import ztest
from statsmodels.stats.proportion import proportions_ztest


In [7]:
col_names = ['id', 'pH', 'Hardness', 'Solids', 'Chloramines', 'Sulfate',
             'Conductivity', 'OrganicCarbon', 'Trihalomethanes', 'Turbidity', 'Potability']
df = pd.read_csv('../data/water_potability.csv', names=col_names)

In [8]:
df.head()

id         pH    Hardness        Solids  Chloramines     Sulfate  \
0   1   8.316766  214.373394  22018.417441     8.059332  356.886136   
1   2   9.092223  181.101509  17978.986339     6.546600  310.135738   
2   3   5.584087  188.313324  28748.687739     7.544869  326.678363   
3   4  10.223862  248.071735  28749.716544     7.513408  393.663396   
4   5   8.635849  203.361523  13672.091764     4.563009  303.309771   

   Conductivity  OrganicCarbon  Trihalomethanes  Turbidity  Potability  
0    363.266516      18.436524       100.341674   4.628771           0  
1    398.410813      11.558279        31.997993   4.075075           0  
2    280.467916       8.399735        54.917862   2.559708           0  
3    283.651634      13.789695        84.603556   2.672989           0  
4    474.607645      12.363817        62.798309   4.401425           0

#### a) Nilai rata - rata pH di atas 7?

$H_0$: Nilai rata rata pH sama dengan $7$ ($\mu = 7$)  
$H_1$: Nilai rata rata pH lebih dari $7$ ($\mu > 7$)

Tingkat Signifikan $\alpha = 0.05$

Lakukan uji statistik dengan one tailed test ke arah kanan (right tailed test) karena $\mu > 7$. Ambil critical section $z > z_\alpha$

Hitung nilai z dengan rumus

$$ z = \frac{\bar{x} - \mu}{\sigma/\sqrt{n}} $$

Tolak $H_0$ jika $z > z_\alpha$ dan $p < \alpha$  
Terima $H_0$ jika $z \le z_\alpha$ dan $p \ge \alpha$

In [33]:
miu = 7
alpha = 0.05

#calculate z and p using ztest module
z, p = ztest(df["pH"], value=miu)

#calculate z_alpha
z_a = st.norm.ppf(1-alpha)

print(f"Nilai z: {round(z, 4)}")
print(f"Nilai z_alpha: {round(z_a, 4)}")
print(f"Nilai p: {round(p, 4)}")




Nilai z: 2.4854
Nilai z_alpha: 1.6449
Nilai p: 0.0129


Nilai $z$ lebih besar dibandingkan dengan $z_\alpha$ $(2.4854 > 1.6449)$  
Nilai $p$ lebih kecil dibandingkan $\alpha$  $(0.0129 < 0.05)$  
Maka tolak $H_0$

**Kesimpulan: rata-rata pH lebih dari $7$**

#### b) Nilai rata rata hardness tidak sama dengan 205?

$H_0$: Nilai rata rata Hardness sama dengan $205$ ($\mu = 205$)  
$H_1$: Nilai rata rata Hardness tidak sama dengan $205$ ($\mu \ne 205$)

Tingkat Signifikan $\alpha = 0.05$

Lakukan uji statistik dengan two tailed test karena harus di cek pada bagian kanan $\mu > 205$ dengan $z > z_{\alpha/2}$ dan pada bagian kiri $\mu < 205$ dengan $z < -z_{\alpha/2}$

Hitung nilai z dengan rumus

$$ z = \frac{\bar{x} - \mu}{\sigma/\sqrt{n}} $$

Tolak $H_0$ jika $z > z_{\alpha/2}$ atau $z < -z_{\alpha/2}$ dan $p < \alpha$  
Terima $H_0$ jika $-z_{\alpha/2} \le z \le z_{\alpha/2}$ dan $p \ge \alpha$

In [37]:
miu = 205
alpha = 0.05

#calculate z and p using ztest module
z, p = ztest(df["Hardness"], value=miu)

#calculate z_alpha
z_a = st.norm.ppf(1-alpha/2)

print(f"Nilai z: {round(z, 4)}")
print(f"Nilai z_alpha/2: {round(z_a, 4)}")
print(f"Nilai p: {round(p, 4)}")


Nilai z: -12.4031
Nilai z_alpha/2: 1.96
Nilai p: 0.0


Nilai $z$ lebih kecil dibandingkan dengan $-z_{\alpha/2}$ $(-12.4031 < 1.96)$  
Nilai $p$ lebih kecil dibandingkan $\alpha$  $(0 < 0.05)$  
Maka tolak $H_0$

**Kesimpulan: rata-rata Hardness tidak sama dengan $205$**

#### c) Nilai Rata-rata 100 baris pertama kolom Solids bukan 21900?

$H_0$: Nilai rata rata 100 baris pertama kolom Solids sama dengan $21900$ ($\mu = 21900$)  
$H_1$: Nilai rata rata 100 baris pertama kolom Solids tidak sama dengan $21900$ ($\mu \ne 21900$)

Tingkat Signifikan $\alpha = 0.05$

Lakukan uji statistik dengan two tailed test karena harus di cek pada bagian kanan $\mu > 21900$ dengan $z > z_{\alpha/2}$ dan pada bagian kiri $\mu < 21900$ dengan $z < -z_{\alpha/2}$

Hitung nilai z dengan rumus

$$ z = \frac{\bar{x} - \mu}{\sigma/\sqrt{n}} $$

Tolak $H_0$ jika $z > z_{\alpha/2}$ atau $z < -z_{\alpha/2}$ dan $p < \alpha$  
Terima $H_0$ jika $-z_{\alpha/2} \le z \le z_{\alpha/2}$ dan $p \ge \alpha$

In [39]:
miu = 21900
alpha = 0.05

#calculate z and p using ztest module
z, p = ztest(df["Solids"].head(100), value=miu)

#calculate z_alpha/2
z_a = st.norm.ppf(1-alpha/2)

print(f"Nilai z: {round(z, 4)}")
print(f"Nilai z_alpha/2: {round(z_a, 4)}")
print(f"Nilai p: {round(p, 4)}")

Nilai z: 0.5637
Nilai z_alpha/2: 1.96
Nilai p: 0.573


Nilai $z$ berada pada rentang $-z_{\alpha/2} \le z \le z_{\alpha/2}$ $(-1.96 \le 0.5637 \le 1.96)$  
Nilai $p$ lebih besar dibandingkan $\alpha$  $(0.573 \ge 0.05)$  
Maka terima $H_0$

**Kesimpulan: rata-rata 100 baris pertama kolom solids sama dengan $21900$**

#### d) Proporsi nilai Conductivity yang lebih dari 450, adalah tidak sama dengan 10%?

$H_0$: Proporsi nilai Conductivity yang lebih dari 450 sama dengan $10\%$ ($p = 0.1$)  
$H_1$: Proporsi nilai Conductivity yang lebih dari 450 tidak sama dengan $10\%$ ($p \ne 0.1$)  

Tingkat Signifikan $\alpha = 0.05$

Lakukan uji statistik dengan two tailed test karena harus di cek pada bagian kanan dengan $z > z_{\alpha/2}$ dan pada bagian kiri dengan $z < -z_{\alpha/2}$

Hitung nilai z dengan rumus

$$ z = \frac{\hat{p} - p_0}{\sqrt{p_0q_0/n}} $$

Tolak $H_0$ jika $z > z_{\alpha/2}$ atau $z < -z_{\alpha/2}$ dan $p < \alpha$  
Terima $H_0$ jika $-z_{\alpha/2} \le z \le z_{\alpha/2}$ dan $p \ge \alpha$

In [59]:
p_0 = 0.10
alpha = 0.05

conductivity_over_450 = len(df[df["Conductivity"] > 450])

z, p = proportions_ztest(conductivity_over_450,
                         len(df), value=p_0, prop_var=p_0)
z_a = st.norm.ppf(1-alpha/2)

print(f"Nilai z: {round(z, 4)}")
print(f"Nilai z_alpha/2: {round(z_a, 4)}")
print(f"Nilai p: {round(p, 4)}")


Nilai z: 40.4464
Nilai z_alpha/2: 1.96
Nilai p: 0.0


Nilai $z$ lebih dari $z_{\alpha/2}$ $(40.4464 > 1.96)$  
Nilai $p$ lebih kecil dibandingkan $\alpha$  $(0.0 < 0.05)$  
Maka tolak $H_0$

**Kesimpulan: Proporsi nilai Conductivity yang lebih dari 450 tidak sama dengan $10\%$**

#### e) Proporsi nilai Trihalomethanes yang kurang dari 40, adalah kurang dari 5%

$H_0$: Proporsi nilai Trihalomethanes yang kurang dari 40 sama dengan $5\%$ ($p = 0.05$)  
$H_1$: Proporsi nilai Trihalomethanes yang kurang dari 40 kurang dari $5\%$ ($p < 0.05$)  

Tingkat Signifikan $\alpha = 0.05$

Lakukan uji statistik dengan one tailed test karena harus di cek pada bagian kiri dengan $z < z_{\alpha}$

Hitung nilai z dengan rumus

$$ z = \frac{\hat{p} - p_0}{\sqrt{p_0q_0/n}} $$

Tolak $H_0$ jika $z < z_\alpha$ dan $p < \alpha$  
Terima $H_0$ jika $z \ge z_\alpha$ dan $p \ge \alpha$

In [63]:
p_0 = 0.05
alpha = 0.05

trihalomethanes_less_40 = len(df[df["Trihalomethanes"] < 40])

z, p = proportions_ztest(trihalomethanes_less_40,
                         len(df), value=p_0, prop_var=p_0)
z_a = st.norm.ppf(1-alpha)

print(f"Nilai z: {round(z, 4)}")
print(f"Nilai z_alpha: {round(z_a, 4)}")
print(f"Nilai p: {round(p, 4)}")


Nilai z: 0.5629
Nilai z_alpha: 1.6449
Nilai p: 0.5735


Nilai $z$ lebih kecil dari $z_{\alpha}$ $(0.5629 < 1.6449)$  
Nilai $p$ lebih besar dibandingkan $\alpha$  $(0.5735 > 0.05)$  
Maka terima $H_0$

**Kesimpulan: Proporsi nilai Trihalomethanes yang kurang dari 40 kurang dari $5\%$**